### ISSUE TITLE
Trigger/Content warning component

### ISSUE THREAD

**Issue by EmergencyTemporalShift** [Yellow-Dog-Man/Resonite-Issues#1240]
### Is your feature request related to a problem? Please describe.

People on Resonite have different sensitivities to various things, and as fun as it is to share random worlds or items around randomly, this can have an adverse effect on those people.

### Describe the solution you'd like

 I propose a component with text fields that can list common triggers for people, such as loud noises or sexual content (sexual content or NSFW might want to be handled differently to prevent spawning for underage people, I can make another post later if required). People with those sensitivities should then be able to detect those tags and opt out of using worlds or items. Items could be handled with Protoflux, but worlds would need a more significant change I would imagine.)

### Describe alternatives you've considered

Putting warnings in the name of the item can ruin any fun shock factor and can make the item too wordy.

**Comment by NukiRaccoon** [comment]
The only way I can see this working is if we have some sort of standardized list already set up, so I guess we should make a list of what people are upset by.

Off the top of my head, the kind of content warnings I would add would be...
Guns
Violence
Loud
Surprise
Sex
Nudity
Transformation (as in, this will make a change to your avatar)
Joke
Disruptive
Heavy (as in, heavy load on an older computer)

The other issue I see is that I don't see a way to enforce this, especially considering new users wouldn't know, but I suppose getting the established builders to agree to use this could potentially be enough for the long term, considering the community is pretty tight-knit at this point, and that's who people are learning from.

I would like to request that this only apply when the slot is active, either when it's loaded or when the user tries to activate it, as there are a lot of multi-use items and features which are baked into avatars, so it would be more useful to be able to mark those slots and have it stop a process that tries to activate them.  Eg, if I have some sort of firework launcher on my hand, and I mark the firework as loud and disruptive, it will refuse to activate in sessions where that's not appropriate, but I can still load the avatar.

**Comment by Geenz** [comment]
Stood up #1564.  Closing this in favor of that.
